modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3897,7 +3897,9 @@ def compute_aligned_columns(*dfs, initial_columns=None):
                 # aligned columns
                 parts = result._partitions.flatten()
                 aligned_columns = parts[0].apply(
-                    compute_aligned_columns,
+                    # TODO: unidist on MPI execution requires for this function to be preprocessed,
```

Comment:
We could look at this after https://github.com/modin-project/modin/pull/6900 is merged.

Reply:
@dchigarev, I ran the previously failing test on latest master without preprocessing `compute_aligned_columns` and it worked. Do you think we should revert this PR? At least, remove the TODO comment.

Reply:
Indeed, it seems that #6900 fixed the issue for both dask and unidist.
I'll prepare a PR reverting the change